Fixes to support installation of packer version 0.10.0 and higher

johnmehringer opened the pull request:
Fixed puppet module to support versions 0.10.0 and higher of packer installation. The code to set the packer install package name and
version check command line were doing string comparisons instead of
number comparisons on the major, minor, bug fix subversions.  This
caused the code to configure package 0.10.0 as if it was 0.6.x or
earlier.  The string '0.10' is less than '0.7' but the version
numbers are not.

Fixed the packer install to continue on failure to remove the
old packer binary.  This was preventing the packer install from
completing on a freshly built box.

Commit:
Fixed puppet module to support versions 0.10.0 and higher of packer
installation. The code to set the packer install package name and
version check command line were doing string comparisons instead of
number comparisons on the major, minor, bug fix subversions.  This
caused the code to configure package 0.10.0 as if it was 0.6.x or
earlier.  The string '0.10' is less than '0.7' but the version
numbers are not.

Fixed the packer install to continue on failure to remove the
old packer binary.  This was preventing the packer install from
completing on a freshly built box.

johnmehringer commented:
@liamjbennett Can you take a look at my changes and if they look acceptable  merge them into the opentable repo for puppet-packer?  

The current packer jenkins slaves are broken because the 0.10.0 version of packer failed to download and install.  I have not fully tested this install but I think my changes above should at least allow the new packer version to install.

Thanks!

fessyfoo commented:
suggseting the version stuff use stdlibs versioncmp  https://docs.puppet.com/puppet/latest/reference/function.html#versioncmp

Commit:
Refactored version code to use puppet stdlib versioncmp().
Added a requirement to ensure that unzip is installed first
since the code requires it to function.

johnmehringer commented:
@fessyfoo I refactored the code to use versioncmp. Please take a look and merge if it looks good.

Thanks!
